python/foglamp/common/configuration_manager.py

```diff
@@ -265,13 +278,22 @@ def get_entry_val(k):
 
     async def _create_new_category(self, category_name, category_val, category_description):
         try:
+            if isinstance(category_val, dict):
+                new_category_val = copy.deepcopy(category_val)
+                # Remove "deprecated" items from a new category configuration
+                for i, v in category_val.items():
+                    if 'deprecated' in v:
+                        new_category_val.pop(i)
+            else:
+                new_category_val = category_val
+
             audit = AuditLogger(self._storage)
-            await audit.information('CONAD', {'name': category_name, 'category': category_val})
+            await audit.information('CONAD', {'name': category_name, 'category': new_category_val})
             payload = PayloadBuilder().INSERT(key=category_name, description=category_description,
-                                              value=category_val).payload()
+                                              value=new_category_val).payload()
             result = await self._storage.insert_into_tbl("configuration", payload)
             response = result['response']
-            self._cacheManager.update(category_name, category_val)
+            self._cacheManager.update(category_name, new_category_val)
```

Comment:
We should save the value as saved and returned in the DB.

Reply:
Insert and update operations return just a string - "inserted" or "updated" as response. Hence "response" is not suitable to be added to the cache.

Reply:
Not the response, but we MAY want to re-read the value from DB, immediately after insertion// update; to update cache.

Reply:
Now cache updated from db value in _update_category() and not in _create_new_category() where it is not needed.